hyrisecockpit/api/app/metric/controller.py

```diff
@@ -0,0 +1,88 @@
+"""Controllers for metrics."""
+from flask import request
+from flask_accepts import accepts, responds
+from flask_restx import Namespace, Resource
+
+from .interface import TimeIntervalInterface
+from .schema import (
+    DetailedQueryInformationSchema,
+    LatencySchema,
+    QueueLengthSchema,
+    ThroughputSchema,
+)
+from .service import MetricService
+
+api = Namespace("Metric", description="Metric data.")
+
+
+@api.route("/throughput")
+class ThroughputController(Resource):
+    """Controller of throughput data."""
+
+    @accepts(
+        dict(name="startts", type=int),  # noqa
+        dict(name="endts", type=int),  # noqa
+        dict(name="precision", type=int),  # noqa
+        api=api,
+    )
+    @responds(schema=ThroughputSchema(many=True), api=api)
+    def get(self):
+        """Get throughput data for the requested time interval."""
+        interface: TimeIntervalInterface = TimeIntervalInterface(
+            startts=request.parsed_args["startts"],
+            endts=request.parsed_args["endts"],
+            precision=request.parsed_args["precision"],
+        )
+        return MetricService.get_throughput(interface)
+
+
+@api.route("/latency")
+class LatencyController(Resource):
+    """Controller for latency data."""
+
+    @accepts(
+        dict(name="startts", type=int),  # noqa
+        dict(name="endts", type=int),  # noqa
+        dict(name="precision", type=int),  # noqa
+        api=api,
+    )
+    @responds(schema=LatencySchema(many=True), api=api)
+    def get(self):
+        """Get latency data for the requested time interval."""
+        interface: TimeIntervalInterface = TimeIntervalInterface(
+            startts=request.parsed_args["startts"],
+            endts=request.parsed_args["endts"],
+            precision=request.parsed_args["precision"],
+        )
+        return MetricService.get_latency(interface)
+
+
+@api.route("/queue_length")
+class QueueLengthController(Resource):
+    """Controller for queue length data."""
+
+    @accepts(
+        dict(name="startts", type=int),  # noqa
+        dict(name="endts", type=int),  # noqa
+        dict(name="precision", type=int),  # noqa
+        api=api,
+    )
+    @responds(schema=QueueLengthSchema(many=True), api=api)
+    def get(self):
```

Comment:
If it responds with the `QueueLengthSchema`, I suggest adding a return type with the `QueueLength` model like:
```suggestion
    @responds(schema=QueueLengthSchema(many=True), api=api)
    def get(self) -> List[QueueLength]:
```